moviepy/video/VideoClip.py

```diff
@@ -721,13 +735,25 @@ def set_make_frame(self, mf):
         self.size = self.get_frame(0).shape[:2][::-1]
 
     @outplace
-    def set_audio(self, audioclip):
-        """Attach an AudioClip to the VideoClip.
-
-        Returns a copy of the VideoClip instance, with the `audio`
+    def set_audio(self, audioclip, start_time=0, keep_original=False):
+        """
+        Attach an AudioClip to the VideoClip.
+        :param audioclip: new audioclip
+        :param start_time: some case audio not start at 0s
+        :param keep_original: composite AudioClip
+        :return: Returns a copy of the VideoClip instance, with the `audio`
         attribute set to ``audio``, which must be an AudioClip instance.
         """
-        self.audio = audioclip
+        if start_time > 0:
+            from moviepy.editor import concatenate_audioclips
+            dummy = audioclip.audio_loop(duration=start_time).volumex(0).subclip(0, start_time)
+            audioclip = concatenate_audioclips([dummy, audioclip])
+
+        if self.audio is not None and keep_original:
+            from moviepy.editor import CompositeAudioClip
```

Comment:
You can import this at the top of the file, and don't import from moviepy.editor, use the exact file (moviepy.audio.AudioClip). Same goes for the import on line 748, and any others.

Reply:
ok, I have fix it.